src/main/java/org/acme/infinispanclient/model/HPCharacter.java

```diff
@@ -0,0 +1,70 @@
+package org.acme.infinispanclient.model;
+
+import java.util.Objects;
+
+/**
+ * Harry Potter saga character
+ */
+public class HPCharacter {
+   private final int id;
+   private final String name;
+   private final String bio;
+   private final int type;
+
+   public HPCharacter(int id, String name, String bio, int type) {
+      this.id = id;
+      this.name = name;
+      this.bio = bio;
+      this.type = type;
+   }
+
+   @Override
+   public boolean equals(Object o) {
+      if (this == o) return true;
+      if (o == null || getClass() != o.getClass()) return false;
+      HPCharacter character = (HPCharacter) o;
+      return id == character.id &&
+            Objects.equals(name, character.name) &&
+            Objects.equals(bio, character.bio) &&
+            Objects.equals(type, character.type);
+   }
+
+   @Override
+   public String toString() {
+      return "Character{" +
+            "id=" + id +
+            ", name='" + name + '\'' +
+            ", bio='" + bio + '\'' +
+            ", type='" + type + '\'' +
+            '}';
+   }
+
+   @Override
+   public int hashCode() {
+      return Objects.hash(id, name, bio, type);
+   }
+
+   public int getId() {
+      return id;
+   }
+
+   public String getName() {
+      return name;
+   }
+
+   public String getBio() {
+      return bio;
+   }
+
+   public int getType() {
+      return type;
+   }
+
+   public boolean isAtHogwarts() {
```

Comment:
Being "at hogwarts" is a bit confusing. When I was first reading this I assumed it meant the person was physically there or not, but looking at this it seems to be whether the person is allowed/goes to Hogwarts instead. I would suggest some clarification around this.